Curs3/app/src/main/java/com/example/curs3/MainActivity.java

```diff
@@ -0,0 +1,133 @@
+package com.example.curs3;
+
+import android.os.Bundle;
+import android.util.Patterns;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity {
+    private EditText email;
+    private EditText phone;
+    private CheckBox terms;
+    private Button submit;
+    private TextView user_input;
+
+    private Button next_layout_1_2;
+    private Button next_layout_2_3;
+
+    private Spinner androidVersions;
+    private ArrayList<String> androidVersionsList;
+    private ArrayAdapter<String> spinnerAdapter;
+
+    protected void initSubmitButton() {
+        submit.setOnClickListener(v -> {
+            boolean isValidInput = true;
+
+            String emailStr = email.getText().toString();
+            String phoneStr = phone.getText().toString();
+
+            if (emailStr.isEmpty() ||
```

Comment:
To make the code readable and reusable please consider creating a dedicated method that validates if the a string is a valid email address. Same idea for checking the phone number field